dm/syncer/syncer.go

```diff
@@ -440,11 +441,13 @@ func (s *Syncer) Init(ctx context.Context) (err error) {
 	}
 
 	// when Init syncer, set active relay log info
-	err = s.setInitActiveRelayLog(ctx)
-	if err != nil {
-		return err
+	if s.cfg.Meta == nil || s.cfg.Meta.BinLogName != binlog.FakeBinlogName {
+		err = s.setInitActiveRelayLog(ctx)
```

Comment:
seems `setInitActiveRelayLog` can move to sync.Run so that we don't need this check `s.cfg.Meta.BinLogName != binlog.FakeBinlogName` ?

Reply:
> when Init syncer, set active relay log info

we will risk the relay log being purged between Init and Run. I prefer we don't change old logic if we don't have enough thinking.

Reply:
Init should not have long-running tasks by definition, it will cause whole pipeline timeout.